grass/kernel.c

```diff
@@ -53,6 +53,7 @@ static void proc_try_syscall(struct process *proc);
 static void excp_entry(uint id) {
     if (id >= EXCP_ID_ECALL_U && id <= EXCP_ID_ECALL_M) {
         proc_set[curr_proc_idx].mepc += 4;
+        memcpy(&proc_set[curr_proc_idx].sc, (void *)SYSCALL_ARG, sizeof(struct syscall));
```

Comment:
As mentioned later, after this line, you should add something like `proc_set[curr_proc_idx].syscall.msg.status = PENDING`.